src/modules/auth/decorators/get-account.decorator.ts

```
@@ -0,0 +1,20 @@
+import {
+  createParamDecorator,
+  ExecutionContext,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { RequestAccount } from '../types/request-account.type';
+import { extractRequestAccount } from '#/shared/extractors/request-account.extractor';
+
+export const GetAccount = createParamDecorator(
+  (data: keyof RequestAccount | undefined, ctx: ExecutionContext) => {
+    const request = ctx.switchToHttp().getRequest();
+    const account = extractRequestAccount(request);
+
+    if (!account) {
+      throw new UnauthorizedException();
+    }
+
+    return data ? account[data] : account;
```

What should happen if account[data] is undefined? Let's throw an error here